lib/orm.php

```diff
@@ -2196,6 +2196,82 @@ public function save() {
 		return $success;
 	}
 
+	/**
+	 * Inserts multiple rows in a single query. Expects new rows as it's a strictly insert function, not an update one.
+	 *
+	 * @param array $models Array of model instances to be inserted.
+	 *
+	 * @return bool True for successful insert, false for failed.
+	 *
+	 * @example From the Indexable_Link_Builder class: $this->seo_links_repository->query()->insert_many( $links );
+	 *
+	 * @throws \InvalidArgumentException Invalid instances to be inserted.
+	 * @throws \InvalidArgumentException Instance to be inserted is not a new one.
+	 */
+	public function insert_many( $models ) {
+		// Validate the input first.
+		if ( ! \is_array( $models )  ) {
+			throw new \InvalidArgumentException( 'Invalid instances to be inserted' );
+		}
+
+		if ( empty( $models ) ) {
+			return false;
```

Comment:
one could argue that all work has been completed succesfully. returning FALSE may trigger error handling and user notifications?

Reply:
One would, yeah 😂 

I'm actually torn, as passing an empty array shouldn't happen and we should have never went into the insert_many function in the first place. So in order to inform the calling function that it has passed an empty array, we return false (as true would mean that the instert_many managed to insert entries.

But I can be convinced otherwise :)

Reply:
how do other insert methods deal with empty models? we should do the same.

Reply:
Nice thought :) In update_many() we return `true`, so that's what I'm gonna do here too.